sale_payment_method/sale.py

```diff
@@ -0,0 +1,258 @@
+        return True
+
+    @api.multi
+    def _add_payment(self, journal, amount, date, description=None):
+        """ Generate move lines entries to pay the sale order. """
+        move_model = self.env['account.move']
+        period_model = self.env['account.period']
+        period = period_model.find(dt=date)
+        move_name = description or self._get_payment_move_name(journal, period)
+        move_vals = self._prepare_payment_move(move_name, journal,
+                                               period, date)
+        move_lines = self._prepare_payment_move_lines(move_name, journal,
+                                                      period, amount, date)
+
+        move_vals['line_id'] = [(0, 0, line) for line in move_lines]
+        move_model.create(move_vals)
+
+    @api.model
+    def _get_payment_move_name(self, journal, period):
+        sequence = journal.sequence_id
+        if not sequence:
+            raise exceptions.Warning(_('Please define a sequence on the '
+                                       'journal %s.') % journal.name)
+        if not sequence.active:
+            raise exceptions.Warning(_('Please activate the sequence of the '
+                                       'journal %s.') % journal.name)
+
+        sequence = sequence.with_context(fiscalyear_id=period.fiscalyear_id.id)
+        # next_by_id not compatible with new api
+        sequence_model = self.pool['ir.sequence']
+        name = sequence_model.next_by_id(self.env.cr, self.env.uid,
+                                         sequence.id,
+                                         context=self.env.context)
+        return name
+
+    @api.multi
+    def _prepare_payment_move(self, move_name, journal, period, date):
+        return {'name': move_name,
+                'journal_id': journal.id,
+                'date': date,
+                'ref': self.name,
+                'period_id': period.id,
+                }
+
+    @api.multi
+    def _prepare_payment_move_line(self, move_name, journal, period,
+                                   amount, date):
+        # to remove in v9
+        _logger.warning('Deprecated: _prepare_payment_move_line has been '
+                        'deprecated in favor of _prepare_payment_move_lines')
+        return self._prepare_payment_move_lines(move_name, journal, period,
+                                                amount, date)
+
+    @api.multi
+    def _prepare_payment_move_lines(self, move_name, journal, period,
+                                    amount, date):
+        partner = self.partner_id.commercial_partner_id
+        company = journal.company_id
+
+        currency = self.env['res.currency'].browse()
+        amount_currency = 0.0
+        if journal.currency and journal.currency != company.currency_id:
+            currency = journal.currency
+            company_amount = currency.compute(amount, company.currency_id)
+            amount_currency, amount = amount, company_amount
```

Comment:
No `else` branch, because when there is no currency, the `amount_currency` is 0. The values are initialized before the `if`.